build.py

```diff
@@ -0,0 +1,238 @@
+    exit(2)
+
+
+## Polyfill Python will remove link_to in python 3.12
+if not hasattr(Path, "hardlink_to"):
+
+    def hardlink_to_polyfill(self, target):
+        return target.link_to(self)
+
+    Path.hardlink_to = hardlink_to_polyfill
+
+
+@contextlib.contextmanager
+def in_dir(new_path):
+    oldcwd = os.getcwd()
+    try:
+        os.chdir(new_path)
+        yield
+    finally:
+        os.chdir(oldcwd)
+
+
+def process_file(source, dist_path, export_directory, zip):
+    target = export_directory / dist_path
+    target.parent.mkdir(parents=True, exist_ok=True)
+
+    try:
+        # hardlinks take a fraction of time related to copy&paste
+        # Unfortunately, some platforms don't implement or allow them for regular users (E.g. windows)
+        target.hardlink_to(source)
+        # Don't show the user a hard link is tried when the feature is not implemented
+        printe("debug: hardlinked", source)
+    except NotImplementedError:  # Not implemented in windows
+        printe("debug: copy", source)
+        shutil.copy(source, target)
+    except Exception:
+        # Pretends the hardlink was tried but an error happened
+        printe("debug: hardlink", source)
+        raise
+
+    if zip:
+        if source.suffix == ".js":
+            printe("release: uglify js", source)
+            zip_content = subprocess.check_output(
+                [
+                    "uglifyjs",
+                    "--compress",
+                    "--mangle",
+                    "--no-annotations",
+                    "-c",
+                    "drop_console",
+                    "--",
+                    os.fspath(source.absolute()),
+                ]
+            )
+        else:
+            printe("release: include file", source)
+            zip_content = source.read_bytes()
+
+        zip.writestr(os.fspath(dist_path), zip_content)
+
+
+dir_path = Path(__file__).absolute().parent
+
+with in_dir(dir_path):
+
+    def build():
+        here_path = Path(".")
+
+        # This is slow but there should be very few files, so it should be OK
+        files_to_include = list(
+            itertools.chain(
+                here_path.glob("*.html"),
+                here_path.glob("*.css"),
+                here_path.glob("*.js"),
+                (here_path / "icon").iterdir(),
+            )
+        )
+
+        versions = filter(lambda p: p.is_dir(), VERSIONS_PATH.iterdir())
+        # This could be run just once per program run but it would bring unnecessary complexity to the code below
+        shutil.rmtree(EXPORT_PATH, True)
+
+        for version in versions:
+            export_directory = EXPORT_PATH / version.name
+            export_directory.mkdir(parents=True, exist_ok=True)
+
+            if program_args.production:
+                # Takes quite a while to make new zips. So better do it for final tests or when deploying
+                extension_archive = export_directory / OUTPUT_ARCHIVE_NAME.with_suffix(
```

Comment:
```suggestion
                extension_archive = EXPORT_PATH / OUTPUT_ARCHIVE_NAME.with_suffix(
```
easier to find if its in the main dir imo

Reply:
I don't know which one I prefer, so I'll do that.